### ISSUE TITLE
(With solution ->) Request header field api-request-source is not allowed by Access-Control-Allow-Headers

### ISSUE THREAD

**Issue by protyze** [api-ai/apiai-nodejs-client#70]
Hi there,

The API seems to block requests with the following message: "Request header field api-request-source is not allowed by Access-Control-Allow-Headers"

I saw that in the code here:
https://github.com/api-ai/apiai-nodejs-client/blob/master/module/request.js
on line 55 where the header is set up, the api-request-source header is added:
'api-request-source': self.requestSource

but in the documentation here: https://api.ai/docs/reference/agent/query#post_query

That header isn't there. After removing it from the code of the Node.js SDK, the request worked perfectly.

Should this be removed?

Best,
Jan

**Comment by arjun-coursera** [comment]
Agreed. I had the exact same issue and fixed by removing api-request-source on request.js as mentioned in the above post.

**Comment by samsherwood** [comment]
Just noting that this issue is still open and still affects the Node.js SDK. It's incredibly annoying, as production builds must be edited constantly to accommodate this issue.

**Comment by Matrixzen** [comment]
Is there a fix in the near future?

On Oct 13, 2017, at 1:43 PM, Sam Sherwood <notifications@github.com<mailto:notifications@github.com>> wrote:


Just noting that this issue is still open and still affects the Node.js SDK. It's incredibly annoying, as production builds must be edited constantly to accommodate this issue.
